### PR TITLE
Update srm.py

### PR DESCRIPTION
**Pull request by CameronTEllis**
Added code based on what is in rsrm to allow for transformation of new participant data without changing the shared response s

### CODE REVIEW

**Comment by mihaic** [comment]
@CameronTEllis, the Jenkins Linux failure is unrelated. I am working on it.

We need tests for the new code.

File: brainiak/funcalign/srm.py [hunk 1]
```diff
@@ -357,6 +357,60 @@ def _likelihood(self, chol_sigma_s_rhos, log_det_psi, chol_sigma_s,
...
+        return U.dot(V)
+
+    def transform_subject(self, X):
+        """Transform a new subject using the existing model
+
+        Parameters
+        ----------
+
+        X : 2D array, shape=[voxels, timepoints]
+            The fMRI data of the new subject.
```
Comment (TuKo) [inline brainiak/funcalign/srm.py]: Please add that should be under the same stimuli as the original subjects.

File: brainiak/funcalign/srm.py [hunk 2]
```diff
@@ -357,6 +357,60 @@ def _likelihood(self, chol_sigma_s_rhos, log_det_psi, chol_sigma_s,
...
+        # Check if the model exist
+        if hasattr(self, 'w_') is False:
+            raise NotFittedError("The model fit has not been run yet.")
+
+        # Check the number of TRs in the subject
+        if X.shape[1] != self.s_.shape[1]:
+            raise ValueError("The number of timepoints(TRs) does not match the"
+                             "one in the model.")
+
+        for i in range(self.n_iter):
```
Comment (TuKo) [inline brainiak/funcalign/srm.py]: There is no need for a for here. You should call the function once.

File: brainiak/funcalign/srm.py [hunk 3]
```diff
@@ -678,6 +732,59 @@ def _compute_shared_response(self, data, w):
...
```
Comment (TuKo) [inline brainiak/funcalign/srm.py]: There is no need for a for-loop here. You should call the function once.

File: tests/funcalign/test_srm.py [hunk 4]
```diff
@@ -98,7 +98,14 @@ def test_can_instantiate():
             "Invalid computation of SRM! (wrong # features after transform)")
         assert new_s[subject].shape[1] == samples, (
             "Invalid computation of SRM! (wrong # samples after transform)")
-
+
```
Comment (TuKo) [inline tests/funcalign/test_srm.py]: Let's add this as part of a new test. 
Add a new test function that creates the object and tries to add the new subject

File: tests/funcalign/test_srm.py [hunk 5]
```diff
@@ -117,7 +117,53 @@ def test_can_instantiate():
...
+    W.append(Q)
+    X.append(Q.dot(S) + 0.1*np.random.random((voxels, samples)))
+
+    for subject in range(1, subjects):
+        Q, R = np.linalg.qr(np.random.random((voxels, features)))
+        W.append(Q)
+        X.append(Q.dot(S) + 0.1*np.random.random((voxels, samples)))
+
+    # Check that runs with 2 subject
+    s.fit(X)
```
Comment (TuKo) [inline tests/funcalign/test_srm.py]: You should test that it does not succeed when the model is not computed before you call fit()

**Comment by TuKo** [comment]
👍  Well done! @mihaic you can go ahead.

**Comment by mihaic** [comment]
@CameronTEllis, you can merge this yourself after you get the tests to pass.

**Comment by CameronTEllis** [comment]
Jenkins, retest this please.

**Comment by mihaic** [comment]
@CameronTEllis, unfortunately, Travis does not respond to that command. Can you please login to Travis and check if you can restart the build? (I am trying to determine if Travis understands GitHub permissions for teams, but I couldn't find an answer yet.)

**Comment by mihaic** [comment]
@CameronTEllis, to cover lines 403 and 777, you should call `transform_subject` before fitting (you are calling `transform`).

Also, it seems to me you have the right and were able to rerun the build on Travis. Can you please confirm?

**Comment by CameronTEllis** [comment]
Great, looking good. Unfortunately I still don't have write access to this repo and can't merge the pull request.

**Comment by mihaic** [comment]
My bad. Please try again. And when you write the merge message, have a look at the style of commit messages we use:
https://github.com/brainiak/brainiak/commits/master

**Pull request merged**